charts/redpanda/helpers.go

```diff
@@ -92,7 +92,26 @@ func StatefulSetPodLabelsSelector(dot *helmette.Dot, statefulSet map[string]any)
 		"app.kubernetes.io/name":      Name(dot),
 	}
 
-	return helmette.Merge(commonLabels, defaults)
+	result := helmette.Merge(AdditionalSelectorLabels, defaults)
+	podLabels := StatefulSetPodLabels(dot, statefulSet)
+	if !IsMapSubset(podLabels, result) {
+		panic("error: computed label selector is not a subset of pod labels!")
+	}
+
+	return result
+}
+
+// IsMapSubset with the help of generics, determines if one map is a subset of another.
```

Comment:
super nit: There's no need to mention generics in the comments as the function signature makes that obvious.